src/expressions/dataTypes/casting/tryCastToType.ts

```diff
@@ -150,6 +150,17 @@ function createCastingFunction(from: ValueType, to: ValueType): (value: Value) =
 		});
 	}
 
+	// If the two types are equal, no casting is required and we can just return the value as it is.
+	if (from === to) {
```

Comment:
I'd do this short-cut all the way at the top of this function, to prevent running the 8-ish compares that are above here.